### DISCUSSION TITLE
Do you think checking network resource is always needed?

### DISCUSSION THREAD

**Discussion by shaucky** [airsdk/Adobe-Runtime-Support#2500] [General]
Hello.

These are happening on a web game based on Flash Player.

There's a web game that's been running for 13 years, since 2009. These years, some gamer have replaced links to some of the game's needed resources by modifying the network agent, such as character appearances and animations, and certain interfaces that have been closed by rules, sometimes even local custom SWF files. However the game offical ignored these things.

Is this something that should be considered when doing AIR development? Would you watch out for something like that?

**Comment by henke37** [comment]
This is essentially just the same old issue of the client not being trusted. Design your code with the assumption that the client is a filthy liar that will send anything you let it, and some things that you don't let it.

**Reply by shaucky** [reply]
Yeah, as i known, when designing server-side code we shouldn't believe data from client. But in my cognition it's what server-side do.
 
Should we design code in client-side to prevent users from replacing network resources needed be loaded? It may causes user change appearances or other content in client. And some game or other applications offer some paid resources, this is easily affect the payment.
 
This is just an example of the problems I found with the Flash Player page game.

**Reply by henke37** [reply]
The issue becomes, who protects the protection code? The chain of trust is broken the instant you distrust the system administrator.
